File: androidApp/v2/core/model/src/main/java/com/deadly/v2/core/model/Recording.kt
package com.deadly.v2.core.model

import kotlinx.serialization.Serializable

/**
 * V2 Recording domain model
 * 
 * Represents a specific recording/taping of a Grateful Dead show.
 * Contains basic recording metadata from the database. Additional 
 * metadata (title, tracks, format) will be loaded from Archive.org API.
 */
@Serializable
data class Recording(
    val identifier: String,
    val showId: String,
    val sourceType: RecordingSourceType,
    val rating: Double,
    val reviewCount: Int,
    val taper: String? = null,
    val source: String? = null,
    val lineage: String? = null,
    val sourceTypeString: String? = null
    // Future Archive API integration will add:
    // val title: String - from Archive.org metadata
    // val tracks: List<Track> - from Archive.org files listing  
    // val format: AudioFormat - detected from track analysis
) {
    /**
     * Whether this recording has a meaningful rating
     */
    val hasRating: Boolean
        get() = rating > 0.0 && reviewCount > 0
    
    /**
     * Formatted rating display
     */
    val displayRating: String
        get() = if (hasRating) "%.1f/5.0 (%d reviews)".format(rating, reviewCount) else "Not Rated"
    
    /**
     * Simple display title using available data
     */
    val displayTitle: String
        get() = "${sourceType.displayName} • $displayRating"
}

/**
 * Recording source types with safe string parsing
 */
@Serializable
enum class RecordingSourceType(val displayName: String) {
    SOUNDBOARD("SBD"),
    AUDIENCE("AUD"), 
    FM("FM"),
    MATRIX("Matrix"),
    REMASTER("Remaster"),
    UNKNOWN("Unknown");
    
    companion object {
        /**
         * Parse source type from string with safe fallback
         */
        fun fromString(value: String?): RecordingSourceType {
            return when (value?.uppercase()) {
                "SBD", "SOUNDBOARD" -> SOUNDBOARD
                "AUD", "AUDIENCE" -> AUDIENCE
                "FM" -> FM
                "MATRIX", "MTX" -> MATRIX
                "REMASTER" -> REMASTER
                else -> UNKNOWN
            }
        }
    }
}

File: iosApp/deadly/Core/Model/Recording.swift
import Foundation

enum SourceBadgeStyle: String, CaseIterable, Sendable {
    case none = "NONE"
    case short = "SHORT"
    case long = "LONG"
    case icon = "ICON"

    var label: String {
        switch self {
        case .none: return "None"
        case .short: return "Short"
        case .long: return "Long"
        case .icon: return "Icon"
        }
    }

    static func fromString(_ value: String?) -> SourceBadgeStyle {
        allCases.first { $0.rawValue.caseInsensitiveCompare(value ?? "") == .orderedSame } ?? .long
    }
}

enum RecordingSourceType: String, Codable, Sendable, Equatable, CaseIterable {
    case soundboard = "SOUNDBOARD"
    case audience = "AUDIENCE"
    case fm = "FM"
    case matrix = "MATRIX"
    case remaster = "REMASTER"
    case unknown = "UNKNOWN"

    var displayName: String {
        switch self {
        case .soundboard: return "SBD"
        case .audience: return "AUD"
        case .fm: return "FM"
        case .matrix: return "Matrix"
        case .remaster: return "Remaster"
        case .unknown: return "Unknown"
        }
    }

    var badgeLabel: String {
        switch self {
        case .soundboard: return "S"
        case .audience: return "A"
        case .fm: return "FM"
        case .matrix: return "M"
        case .remaster: return "R"
        case .unknown: return ""
        }
    }

    var sfSymbolName: String? {
        switch self {
        case .soundboard: return "slider.horizontal.3"
        case .audience: return "recordingtape"
        case .fm: return "radio"
        case .matrix: return "waveform.path"
        case .remaster: return "wand.and.stars"
        case .unknown: return nil
        }
    }

    static func fromString(_ value: String?) -> RecordingSourceType {
        switch value?.uppercased() {
        case "SBD", "SOUNDBOARD": return .soundboard
        case "AUD", "AUDIENCE": return .audience
        case "FM": return .fm
        case "MATRIX", "MTX": return .matrix
        case "REMASTER": return .remaster
        default: return .unknown
        }
    }
}

struct Recording: Codable, Sendable, Equatable, Identifiable {
    let identifier: String
    let showId: String
    let sourceType: RecordingSourceType
    let rating: Double
    let reviewCount: Int
    let taper: String?
    let source: String?
    let lineage: String?
    let sourceTypeString: String?

    var id: String { identifier }

    var hasRating: Bool { rating > 0.0 && reviewCount > 0 }
    var displayRating: String {
        hasRating ? String(format: "%.1f/5.0 (%d reviews)", rating, reviewCount) : "Not Rated"
    }
    var displayTitle: String { "\(sourceType.displayName) • \(displayRating)" }
}
